numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3255,6 +3255,7 @@ array_datetime_data(PyObject *NPY_UNUSED(dummy), PyObject *args)
     }
 
     meta = get_datetime_metadata_from_dtype(dtype);
+    Py_DECREF(dtype);    
```

Comment:
This is probably one of the few ones that could hurt in practice, if someone prints a lot of datetime objects for some reason.